src/sidebar/services/groups.js

```diff
@@ -193,15 +195,27 @@ export default function groups(
    * The groups that are fetched depend on the current user, the URI of
    * the current document, and the direct-linked group and/or annotation.
    *
+   * On startup, `load()` must be called to trigger the initial groups fetch.
+   * Subsequently groups are automatically reloaded if the logged-in user or
+   * main document URI changes.
+   *
    * @return {Promise<Group[]>}
    */
   async function load() {
     // Step 1: Get the URI of the active document, so we can fetch groups
     // associated with that document.
+    let documentUri;
     if (isSidebar) {
       documentUri = await getDocumentUriForGroupSearch();
     }
 
+    if (!didSubscribeToUriChanges) {
```

Comment:
I didn't have any plans to change this in future. It might be worth separating the process of performing the initial group load and subscribing to future frame URI / user profile changes into separate methods.